src/Symfony/Component/Messenger/EventListener/SendFailedMessageToFailureTransportListener.php

```diff
@@ -27,12 +28,14 @@
  */
 class SendFailedMessageToFailureTransportListener implements EventSubscriberInterface
 {
-    private $failureSender;
+    private $failureSenders;
+    private $globalFailureSender;
     private $logger;
 
-    public function __construct(SenderInterface $failureSender, LoggerInterface $logger = null)
+    public function __construct(?SenderInterface $globalFailureSender, LoggerInterface $logger = null, ?ServiceLocator $failureSenders = null)
```

Comment:
I will need the `globalFailureSender` here, because `failureSenders` is map where key is the transport name. 
When setting a global `failure_transport` is not associated to a transport name.

I need it, because in https://github.com/symfony/symfony/pull/34979/files#diff-2eca49162e4b0a72c3825b50f160d944R72 we will ask the failure sender from the transport the event was consumed.

Reply:
Hmm. But could we make `$failureSenders` a FULL array of "receivers' => $failureSenderService.

Take this pseudo-config:

```yml
failure_transport: failure1
transports:
    async1: null
    async2:
       failure_transport: failure2
    failure1: null
    failure2: null
```

In this case, $failureSenders would be a map containing this:

```
[
    async1 => <failure1 sender service>
    async2 => <failure2 sender service>
    failure1 => <failure1 sender service>
    failure2 => <failure2 sender service>
]
```

Then we would have all the info we need in the map, correct? This "occurred" to me by looking *only* at this class (and not thinking about the rest of the code): in this class, it *feels* like we should be able to pass this object all the info it needs with one, full service locator of failure senders.

Reply:
How do you which one is the global one? Because your map just says that one specific transport is assigned to one failure sender service. 

Specify a specific key for global? 

[
    "messenger_global" => <failure1 sender service>
    async1 => <failure1 sender service>
    async2 => <failure2 sender service>
    failure1 => <failure1 sender service>
    failure2 => <failure2 sender service>
]